msdk-rawdata/msdk-rawdata-filters/src/main/java/io/github/msdk/filtering/MSDKFilteringMethod.java

```diff
@@ -1,116 +1,92 @@
-                newScan.setChromatographyInfo(scan.getChromatographyInfo());
-                newScan.setRawDataFile(result);
-                // Store the new data points
-                newScan.setDataPoints(dataPoints.selectDataPoints(mzRange, intensityRange));
-
-                if (canceled) {
-                    return null;
-                }
-                // Add the new scan to the created raw data file
-                result.addScan(newScan);
-            }
-            processedScans++;
-        }
-
-        logger.info("Finished Crop Filter with Raw Data File #"
-                + rawDataFile.getName());
-        return result;
-    }
-
-    @Override
-    public RawDataFile getResult() {
-        return result;
-    }
-
-}
+/* 
+ * (C) Copyright 2015 by MSDK Development Team
+ *
+ * This software is dual-licensed under either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package io.github.msdk.filtering;
+
+import io.github.msdk.MSDKException;
+import io.github.msdk.MSDKMethod;
+import io.github.msdk.datamodel.datapointstore.DataPointStoreFactory;
+import io.github.msdk.datamodel.impl.MSDKObjectBuilder;
+import io.github.msdk.datamodel.msspectra.MsSpectrum;
+import io.github.msdk.datamodel.rawdata.MsScan;
+import io.github.msdk.datamodel.rawdata.RawDataFile;
+import java.util.List;
+import javax.annotation.Nonnull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MSDKFilteringMethod implements MSDKMethod<RawDataFile> {
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    private final @Nonnull MSDKFilteringAlgorithm filteringAlgorithm;
+    private final @Nonnull RawDataFile rawDataFile;
+
+    private int processedScans = 0, totalScans = 0;
+    private RawDataFile result;
+    private boolean canceled = false;
+
+    public MSDKFilteringMethod(@Nonnull RawDataFile rawDataFile, @Nonnull MSDKFilteringAlgorithm filteringAlgorithm) {
+        this.filteringAlgorithm = filteringAlgorithm;
+        this.rawDataFile = rawDataFile;
+    }
+
+    @Override
+    public Float getFinishedPercentage() {
+        if (totalScans == 0) {
+            return null;
+        } else {
+            return (float) processedScans / totalScans;
+        }
+    }
+
+    @Override
+    public RawDataFile execute() throws MSDKException {
+        logger.info("Started Filter " + this.filteringAlgorithm.getName() + " with Raw Data File #"
+                + rawDataFile.getName());
+
+        // Create a new raw data file
+        result = MSDKObjectBuilder.getRawDataFile(this.rawDataFile.getName(), this.rawDataFile.getOriginalFile(), this.rawDataFile.getRawDataFileType(), DataPointStoreFactory.getMemoryDataStore());

```

Comment:
The data point store should be a parameter of this method (in constructor).
If you hardcode in ```DataPointStoreFactory.getMemoryDataStore()``` then MSDK will run out of memory very quickly.